docker/vcf_combiner/usr/bin/combine_vcf.py

```diff
@@ -0,0 +1,271 @@
+import logging
+import uuid
+from collections import namedtuple
+from collections import OrderedDict
+import types
+import vcf
+
+Version = '1.2'
+ProductName = os.path.basename(__file__)
+
+COL_PARTITION_SIZE_UNIT = 16384
+DefaultVIDFile = "/usr/share/cont-intel/vid.json"
+DefaultVIDEnv = 'HOST_VID_PATH'                         # to run the script at host, specify the path to a vid file
+logging.basicConfig(stream=sys.stdout, level=logging.INFO)
+
+##### loader config file tags
+loader_cfg_t = {
+    "produce_combined_vcf": True,
+    "num_parallel_vcf_files": 1,
+    "callset_mapping_file" : "",
+    "vcf_output_format" : "z",
+    "vcf_output_filename" : "",
+    "index_output_VCF": True,
+    "reference_genome" : "",
+    "column_partitions" : [],
+    "do_ping_pong_buffering" : True,
+    "offload_vcf_output_processing" : True,
+    "produce_GT_field" : False,
+    "size_per_column_partition" : COL_PARTITION_SIZE_UNIT,
+    "vid_mapping_file": ""
+}
+cp_pos = namedtuple('pos_only', "begin, vcf_output_filename")
+cp_chr = namedtuple('chromosome', 'begin, end, vcf_output_filename')
+##### loader config file tags
+
+def get_loader_cfg(**kwargs):
+    ''' named tuple loader config '''
+    f = lambda x: x() if isinstance(x, types.FunctionType) else x
+    updated = {k : f(kwargs[k]) if k in kwargs else v for k, v in loader_cfg_t.items()}
+    np_LoaderCfg = namedtuple('loader_cfg', ','.join(updated.keys()))
+    return np_LoaderCfg(**updated)
+
+def get_col_partition(output_fn, begin, chromosome=None, end=None):
+    cp_nt = cp_pos(begin=begin, vcf_output_filename=output_fn) if not chromosome else \
+        cp_chr(begin={chromosome:begin}, end={chromosome:end}, vcf_output_filename=output_fn)
+    return cp_nt._asdict()
+
+class CombineVCFException(Exception):
+    pass
+
+class CombineVCF(object):
+    ''' VCF file combiner '''
+    logger = logging.getLogger("CombineVCF")
+    brief_options = "i:o:R:c:p"
+    full_options = ['samples=', 'output=', 'reference=', 'callsets=', 'produce_GT_field', \
+            'chromosome=', 'begin=', 'end=', 'dryrun', 'version']
+
+    def __init__(self):
+        self.dryrun = False
+        self.output_file = None
+
+    def _parse_args(self, args):
+        def get_vid_mapping_file():
+            return DefaultVIDFile if os.path.isfile(DefaultVIDFile) else os.environ[DefaultVIDEnv]
+        vmf_args = {"vid_mapping_file": get_vid_mapping_file}
+        def check_chromosome():
+            assert begin, 'missing begin position of chromosome'
+            if end:
+                assert end >= begin, 'chromosome end position must great or equal to begin position'
+            return get_col_partition(self.output_file, begin, chromosome, end if end else None)
+
+        myopts, _ = getopt(args, self.brief_options, self.full_options)
+        callset_mapping_file = None
+        produce_GT_field = False
+        chromosome = None
+        begin = None
+        end = None
+        for opt, user_input in myopts:
```

Comment:
we need to merge the code into master branch as soon as possible because the vcf_combiner is used by other group and there are GenomicsDB checked in this branch. Will add the usage later. Create a Jira task 506.